tests/regression_tests/dagmc/external/main.cpp

```diff
@@ -0,0 +1,94 @@
+#include "openmc/capi.h"
+#include "openmc/cross_sections.h"
+#include "openmc/dagmc.h"
+#include "openmc/error.h"
+#include "openmc/geometry.h"
+#include "openmc/geometry_aux.h"
+#include "openmc/material.h"
+#include "openmc/nuclide.h"
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+  using namespace openmc;
+  int openmc_err;
+
+  // Initialise OpenMC
+  openmc_err = openmc_init(argc, argv, nullptr);
+  if (openmc_err == -1) {
+    // This happens for the -h and -v flags
+    return EXIT_SUCCESS;
+  } else if (openmc_err) {
+    fatal_error(openmc_err_msg);
+  }
+
+  // Create DAGMC ptr
+  std::string filename = "dagmc.h5m";
+  std::shared_ptr<moab::DagMC> dag_ptr = std::make_shared<moab::DagMC>();
+  moab::ErrorCode rval = dag_ptr->load_file(filename.c_str());
+  if (rval != moab::MB_SUCCESS) {
+    fatal_error("Failed to load file");
+  }
+
+  // Initialize acceleration data structures
+  rval = dag_ptr->init_OBBTree();
+  if (rval != moab::MB_SUCCESS) {
+    fatal_error("Failed to initialise OBB tree");
+  }
+
+  // Get rid of existing geometry
+  std::unordered_map<std::string, int> nuclide_map_copy =
+    openmc::data::nuclide_map;
+  openmc::data::nuclides.clear();
+  openmc::data::nuclide_map = nuclide_map_copy;
+  openmc::model::surfaces.clear();
+  openmc::model::surface_map.clear();
+  openmc::model::cells.clear();
+  openmc::model::cell_map.clear();
+  openmc::model::universes.clear();
+  openmc::model::universe_map.clear();
+
+  // Update materials (emulate an external program)
+  for (auto& mat_ptr : openmc::model::materials) {
+    mat_ptr->set_temperature(300);
+  }
+
+  // Create new DAGMC universe
+  openmc::model::universes.push_back(
+    std::make_unique<openmc::DAGUniverse>(dag_ptr, ""));
```

Comment:
This call is a little awkward with the extra empty string argument. Can we give `filename` a default value of `""`?

Reply:
done